frontend/apps/reader/modules/readergesture.lua

```diff
@@ -345,39 +369,61 @@ function ReaderGesture:buildMultiswipeMenu()
         table.insert(multiswipes, v)
     end
 
-    if custom_multiswipes_table then
+    if custom_multiswipes_table and #custom_multiswipes_table > 0 then
+        table.insert(multiswipes, true) -- add separator
         for k, v in pairs(custom_multiswipes_table) do
             table.insert(multiswipes, v)
         end
     end
 
     for i=1, #multiswipes do
-        local multiswipe = multiswipes[i]
-        local friendly_multiswipe_name = self:friendlyMultiswipeName(multiswipe)
-        local safe_multiswipe_name = "multiswipe_"..self:safeMultiswipeName(multiswipe)
-        local default_action = self.default_gesture[safe_multiswipe_name] and self.default_gesture[safe_multiswipe_name] or "nothing"
-        table.insert(menu, {
-            text_func = function()
-                local action_name = gesture_manager[safe_multiswipe_name] ~= "nothing" and action_strings[gesture_manager[safe_multiswipe_name]] or _("Available")
-                return T(_("%1   (%2)"), friendly_multiswipe_name, action_name)
-            end,
-            sub_item_table = self:buildMenu(safe_multiswipe_name, default_action),
-            hold_callback = function(touchmenu_instance)
-                if i > #default_multiswipes then
-                    UIManager:show(ConfirmBox:new{
-                        text = T(_("Remove custom multiswipe %1?"), friendly_multiswipe_name),
-                        ok_text = _("Remove"),
-                        ok_callback = function()
-                            -- multiswipes are a combined table, first defaults, then custom
-                            -- so the right index is minus #defalt_multiswipes
-                            custom_multiswipes:removeTableItem("multiswipes", i-#default_multiswipes)
-                            touchmenu_instance.item_table = self:buildMultiswipeMenu()
-                            touchmenu_instance:updateItems()
-                        end,
-                    })
-                end
-            end,
-        })
+        local separator = false
+        if i < #multiswipes and multiswipes[i+1] == true then
+            separator = true
+        end
+        if type(multiswipes[i]) == "string" then -- skip separators (true)
```

Comment:
Or just `if type(multiswipes[i]) ~= "string" then return end`?

Reply:
I don't want to `return` :) But I would have used `continue` instead, if Lua had provided that useful construct, instead of needing imbricated `if end`...
I just want to skip the `true` entry, as it was carried to the previous item.

Reply:
Oh, you can actually do that (sort of).

```lua
if type(multiswipes[i]) ~= "string" then goto continue end
-- the whole shebang
::continue
```

The *bad* part of this solution is that you need to keep coming up with new names for the darn things if you have even a little bit of nesting.

Reply:
Well, not fond of such constructs :)
(to review the small changes induced by this added if indentation, you can always use Diff settings >  Hide whitespace changes)

Reply:
I didn't know about that, thanks for the tip. I wasn't talking about the diff though, although it is indeed easier to focus on the code that way. :-) Like you said, I think it'd be cleaner with a continue like in, C, PHP, Python, Java, etc.

(Anyway, it's good.)